src/runtime/slot-proxy.ts

```diff
@@ -67,4 +67,16 @@ export class SlotProxy {
       handler(event);
     }
   }
+
+  /**
+   * gives all rendering capabilities from the current particle to the given one
+   */
+  rewire(particle: Particle): void {
```

Comment:
Seems worth adding unit tests for this method.

Reply:
Should I create a new file for this say `slot-proxy-test.ts` since there aren't any? Or is there a better place to create these unit tests?

Reply:
Feel free to create a slot-proxy-test.ts file, but I'm not sure how easy it will be to test this method in isolation - feel free to give it a shot, but you may find it's hard to set up.

Another way to go could be to instantiate a running arc (as many tests already do), and then call reloadParticle on the pec, and then assert that a lot of things are happening: Handles are disabled, slot proxies are removed and re-added to a new particle etc.

Reply:
Sounds good! I'll make this test in another PR :D